<?xml version="1.0"?>
<launch>

    <arg name="namespace" default="robofit" />
    <arg name="image_topic" default="video_feedback" />
    <arg name="camera_id" default="0" />
    <arg name="feedback_topic" default="/feedback" />
    <arg name="exercise_topic" default="/exercise" />

    <!-- launch sound_play and speech node -->
    <include file="$(find robofit)/launch/sound_streamer.launch" >
        <arg name="feedback" value="$(arg feedback_topic)" />
        <arg name="namespace" value="$(arg namespace)" />
    </include>

    <!-- show visual feedback -->
    <include file="$(find robofit)/launch/video_streamer.launch" >
        <arg name="image_topic" value="$(arg image_topic)"/>
        <arg name="camera_id" value="$(arg camera_id)"/>
    </include>

    <!-- observe gestures and exercise execution -->
    <include file="$(find myo_ros)/launch/myo_launch.launch" >
        <arg name="exercise_topic" value="$(arg exercise_topic)" />
        <arg name="namespace" value="$(arg namespace)" />
    </include>

    <include file="$(find robofit)/launch/act_mon.launch" >
        <arg name="namespace" value="$(arg namespace)" />
        <arg name="feedback_topic" value="$(arg feedback_topic)" />
        <arg name="exercise_topic" value="$(arg exercise_topic)" />
    </include>


    <!-- state machine for publishing visual and audio feedback -->



</launch>
